moths/data_module.py

```diff
@@ -0,0 +1,155 @@
+            )
+
+    def prepare_data(self):
+        valid_data_path = resolve_config_path(self.config.valid_path_file)
+        log.info(f"Using valid data {str(valid_data_path)} ...")
+
+        with valid_data_path.open("r") as f:
+            self.valid_paths = set([p.strip() for p in f.readlines()])
+
+        label_map_path = resolve_config_path(self.config.label_map_file)
+        log.info(f"Using label map {str(label_map_path)} ...")
+
+        with label_map_path.open("r") as f:
+            lines = [line.strip() for line in f.readlines()]
+
+        self.label_map = {l.split(",")[1]: int(l.split(",")[0]) for l in lines}
+        self.num_classes = len(self.label_map)
+        self.class_names = [l.split(",")[1] for l in lines]
+
+    def _full_dataset(self, transform: Optional[Callable]) -> ConcatImageFolderDataset:
+        datasets = []
+
+        if self.config.fake_data:
+            datasets = [
+                FakeData(transform=transform, num_classes=self.num_classes)
+                for _ in range(3)
+            ]
+        else:
+            for data_path in self.config.data_paths:
+                data_path = resolve_config_path(data_path)
+                log.debug(f"Scanning {str(data_path)} ...")
+                ds = LabelMapImageFolder(
+                    str(data_path),
+                    label_map=self.label_map,
+                    is_valid_file=lambda p: abspath(p) in self.valid_paths,
+                    transform=transform,
+                )
+                datasets.append(ds)
+
+        return ConcatImageFolderDataset(datasets=datasets)
+
+    def setup(self, stage: Optional[str] = None):
+        full_train_dataset = self._full_dataset(self._train_transforms)
+        full_test_dataset = self._full_dataset(self._test_transforms)
+
+        all_indices = list(range(len(full_train_dataset)))
+
+        val_test_fraction = 2 * self.config.test_fraction
+        train_fraction = 1 - val_test_fraction
+
+        targets = full_train_dataset.targets
+
+        train_indices, val_test_indices = train_test_split(
+            all_indices,
+            train_size=train_fraction,
+            test_size=val_test_fraction,
+            stratify=targets,
+        )
+
+        val_test_targets = [targets[x] for x in val_test_indices]
+
+        val_indices, test_indices = train_test_split(
+            val_test_indices,
+            train_size=0.5,
+            test_size=0.5,
+            stratify=val_test_targets,
+        )
+
+        self.train_dataset = Subset(full_train_dataset, train_indices)
+        self.val_dataset = Subset(full_test_dataset, val_indices)
+        self.test_dataset = Subset(full_test_dataset, test_indices)
+
+    def train_dataloader(self):
+        return DataLoader(
+            self.train_dataset,
+            batch_size=self.config.batch_size,
+            num_workers=self.config.num_workers,
+            pin_memory=self.config.pin_memory,
+            shuffle=True,
```

Comment:
Is this also seeded?

Reply:
Yes, pytorch lignthing has `seed_everything` which should seed numpy, pytorch, and lightning all for everything.